src/app/pages/waiting-room-to-car/cat-adi-part2/waiting-room-to-car.cat-adi-part2.page.ts

```diff
@@ -1,30 +1,321 @@
+import { FaultCountProvider } from '@providers/fault-count/fault-count';
+import { getTestData } from '@store/tests/test-data/cat-adi-part2/test-data.cat-adi-part2.reducer';
+import { PersistTests } from '@store/tests/tests.actions';
+import { VehicleChecksQuestion } from '@providers/question/vehicle-checks-question.model';
 import { TestCategory } from '@dvsa/mes-test-schema/category-definitions/common/test-category';
+import { CatADI2UniqueTypes } from '@dvsa/mes-test-schema/categories/ADI2';
+import {
+  OrditTrainedChanged,
+  TrainerRegistrationNumberChanged,
+  TrainingRecordsChanged,
+} from '@store/tests/trainer-details/cat-adi-part2/trainer-details.cat-adi-part2.actions';
+import { getTrainerDetails } from '@store/tests/trainer-details/cat-adi-part2/trainer-details.cat-adi-part2.reducer';
+import {
+  getOrditTrained,
+  getTrainerRegistrationNumber,
+  getTrainingRecords,
+} from '@store/tests/trainer-details/cat-adi-part2/trainer-details.cat-adi-part2.selector';
+import {
+  WaitingRoomToCarBasePageComponent,
+} from '@shared/classes/test-flow-base-pages/waiting-room-to-car/waiting-room-to-car-base-page';
+import { Router } from '@angular/router';
 import { RouteByCategoryProvider } from '@providers/route-by-category/route-by-category';
-import { TestFlowPageNames } from '@pages/page-names.constants';
+import { TestFlowPageNames } from '../../page-names.constants';
+import { WaitingRoomToCarValidationError } from '../waiting-room-to-car.actions';
+import * as waitingRoomToCarActions from '../waiting-room-to-car.actions';
+
+interface WaitingRoomToCarPageState {
+  candidateName$: Observable<string>;
+  registrationNumber$: Observable<string>;
+  transmission$: Observable<GearboxCategory>;
+  schoolCar$: Observable<boolean>;
+  dualControls$: Observable<boolean>;
+  instructorAccompaniment$: Observable<boolean>;
+  supervisorAccompaniment$: Observable<boolean>;
+  otherAccompaniment$: Observable<boolean>;
+  interpreterAccompaniment$: Observable<boolean>;
+  eyesightTestComplete$: Observable<boolean>;
+  eyesightTestFailed$: Observable<boolean>;
+  gearboxAutomaticRadioChecked$: Observable<boolean>;
+  gearboxManualRadioChecked$: Observable<boolean>;
+  vehicleChecksScore$: Observable<VehicleChecksScore>;
+  vehicleChecks$: Observable<CatADI2UniqueTypes.VehicleChecks>;
+  orditTrained$: Observable<boolean>;
+  trainingRecords$: Observable<boolean>;
+  trainerRegistrationNumber$: Observable<number>;
+}
 
 @Component({
-  selector: 'app-waiting-room-to-car-cat-adi-part2',
-  templateUrl: './waiting-room-to-car.cat-adi-part2.page.html',
-  styleUrls: ['./waiting-room-to-car.cat-adi-part2.page.scss'],
+  selector: '.waiting-room-to-car-cat-adi-part2-page',
+  templateUrl: 'waiting-room-to-car.cat-adi-part2.page.html',
 })
-export class WaitingRoomToCarCatAdiPart2Page implements OnInit {
+export class WaitingRoomToCarCatADIPart2Page extends WaitingRoomToCarBasePageComponent implements OnInit {
```

Comment:
Some of the methods here can be removed as they will already live in base page, but we can go through that at the end 👍